tests/dump_tests/module_tests/vlan_test.py

```diff
@@ -0,0 +1,194 @@
+        params["namespace"] = ""
+        m_vlan = Vlan(match_engine)
+        returned = m_vlan.execute(params)
+        expect = create_template_dict(dbs=["CONFIG_DB", "APPL_DB", "ASIC_DB", "STATE_DB"])
+        expect["CONFIG_DB"]["tables_not_found"].append("VLAN")
+        expect["CONFIG_DB"]["tables_not_found"].append("VLAN_MEMBER")
+        expect["APPL_DB"]["tables_not_found"].append("VLAN_TABLE")
+        expect["APPL_DB"]["tables_not_found"].append("VLAN_MEMBER_TABLE")
+        expect["STATE_DB"]["tables_not_found"].append("VLAN_TABLE")
+        expect["STATE_DB"]["tables_not_found"].append("VLAN_MEMBER_TABLE")
+        expect["ASIC_DB"]["tables_not_found"].append("ASIC_STATE:SAI_OBJECT_TYPE_VLAN")
+        expect["ASIC_DB"]["tables_not_found"].append("ASIC_STATE:SAI_OBJECT_TYPE_VLAN_MEMBER")
+        ddiff = DeepDiff(returned, expect, ignore_order=True)
+        assert not ddiff, ddiff
+    
+    def test_unconfigured_vlan(self, match_engine):
+        params = {}
+        params[Vlan.ARG_NAME] = "Vlan5"
+        params["namespace"] = ""
+        m_vlan = Vlan(match_engine)
+        returned = m_vlan.execute(params)
+        expect = create_template_dict(dbs=["CONFIG_DB", "APPL_DB", "ASIC_DB", "STATE_DB"])
+        expect["CONFIG_DB"]["tables_not_found"].append("VLAN")
+        expect["CONFIG_DB"]["tables_not_found"].append("VLAN_MEMBER")
+        expect["APPL_DB"]["tables_not_found"].append("VLAN_TABLE")
+        expect["APPL_DB"]["tables_not_found"].append("VLAN_MEMBER_TABLE")
+        expect["STATE_DB"]["tables_not_found"].append("VLAN_TABLE")
+        expect["STATE_DB"]["tables_not_found"].append("VLAN_MEMBER_TABLE")
+        expect["ASIC_DB"]["tables_not_found"].append("ASIC_STATE:SAI_OBJECT_TYPE_VLAN")
+        expect["ASIC_DB"]["tables_not_found"].append("ASIC_STATE:SAI_OBJECT_TYPE_VLAN_MEMBER")
+        ddiff = DeepDiff(returned, expect, ignore_order=True)
+        assert not ddiff, ddiff
+    
+    def test_garbage_vlan(self, match_engine):
+        params = {}
+        params[Vlan.ARG_NAME] = "garbage"
+        params["namespace"] = ""
+        m_vlan = Vlan(match_engine)
+        returned = m_vlan.execute(params)
+        expect = create_template_dict(dbs=["CONFIG_DB", "APPL_DB", "ASIC_DB", "STATE_DB"])
+        expect["CONFIG_DB"]["tables_not_found"].append("VLAN")
+        expect["CONFIG_DB"]["tables_not_found"].append("VLAN_MEMBER")
+        expect["APPL_DB"]["tables_not_found"].append("VLAN_TABLE")
+        expect["APPL_DB"]["tables_not_found"].append("VLAN_MEMBER_TABLE")
+        expect["STATE_DB"]["tables_not_found"].append("VLAN_TABLE")
+        expect["STATE_DB"]["tables_not_found"].append("VLAN_MEMBER_TABLE")
+        expect["ASIC_DB"]["tables_not_found"].append("ASIC_STATE:SAI_OBJECT_TYPE_VLAN")
+        expect["ASIC_DB"]["tables_not_found"].append("ASIC_STATE:SAI_OBJECT_TYPE_VLAN_MEMBER")
+        ddiff = DeepDiff(returned, expect, ignore_order=True)
+        assert not ddiff, ddiff
+    
```

Comment:
Can you please add test case to test missing specific DB? For e.g. config_db and app_db might be present but ASIC DB might be missing due to some software bugs or other dependencies. Likewise stateDB can also be missing.

Reply:
Done.  I added several new tests.  As I tested this on a different switch from the original one, I updated all the mock OIDs to make them a consistent set.